### ISSUE TITLE
process.stdin cannot be used

### ISSUE THREAD

**Issue by dacap** [nwjs/nw.js#586]
I'm trying to create a window that can receive input (messages) from stdin, but when I use `process.stdin` (or `process.openStdin()`) I receive the following error:

```
Error: Implement me. Unknown stdin file type!
```
##

**Comment by drown-zed** [comment]
Was there any resolution for this?

**Comment by apoco** [comment]
+1. I may be encountering this. I don't get that original error, but `process.stdin` doesn't seem to receive stdin. For example, when running:

```bash
echo 'Hello' | nw .
```

...stdin just gets an `end` event. Sample reading code that just outputs `EOF`:

```javascript
  const { stdin, stdout } = process;

  stdin.resume();
  stdin.setEncoding('utf8');
  stdin
    .on('data', str => {
      stdout.write('Got ' + str);
    })
    .on('end', () => {
      stdout.write('EOF');
    });
```

**Comment by rsrini7** [comment]
I just saw this. 
http://tangiblejs.com/posts/nw-js-johnny-five-arduino-wicked-trio

npm install nwjs-j5-fix

**Comment by dev-mraj** [comment]
+1

**Comment by xtecox** [comment]
+1

**Comment by adibas03** [comment]
+1

**Comment by derrickb** [comment]
+1

Any word from the team on this?  It's been open for a long time.

**Comment by chlp** [comment]
We can try to bypass using https://www.npmjs.com/package/nwjs-j5-fix

**Comment by cotejp** [comment]
> We can try to bypass using https://www.npmjs.com/package/nwjs-j5-fix

I created that package but I really wish I could stop using it. Can we hope for this issue to be bumped up the priority list anytime soon @rogerwang ?

**Comment by Christywl** [comment]
Can reproduce this issue on nwjs v0.26.6.

**Comment by derrickb** [comment]
Can reproduce on 0.27.1 on Mac, and what's worse, @cotejp's fix does not appear to work so there aren't any viable options I can find without looking at other methods for communicating with the app.

**Comment by cotejp** [comment]
> and what's worse, @cotejp's fix does not appear to work

Your use case might be different from mine, but my little fix is still working for me in version 0.27.1.

**Comment by derrickb** [comment]
Ah interesting.  Well I'm glad to know that it should still be working.  Would you mind posting a small snippet showing the most minimal amount of code necessary to test the fix?  Do you use the readline module?  What I've been trying, just to see if I can get any input at all:
```
nw.require('nwjs-j5-fix').fix();
var readline = nw.require('readline');
var rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
  terminal: false
});
rl.on('line', function (line) {
  alert(line);
});
```

My use case is I'm launching nwjs from an Adobe AIR project (probably not a typical use case, but alas...), as a [NativeProcess](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/desktop/NativeProcess.html), and am writing to the process's stdin like so:
`process.standardInput.writeUTF("hello")`

But haven't been able to get any indication that the input as received.  I can retrieve the stdout from this process just fine.  Immediately after launching the process, I do get a "standard error data" event but am not sure what exactly is triggering that.

**Comment by cotejp** [comment]
The nwjs-j5-fix module fixes the issue of NW.js and Johnny-Five not being able to communicate. It does that successfully so I have not tested it under other scenarios. 

You can look at the [source code](https://github.com/cotejp/nwjs-j5-fix/blob/master/nwjs-j5-fix.js) if you wnat to know more. As you'll see, it is **very** simple. I did not even bother with stdout because it is not needed in my scenario.

Cheers!

**Comment by derrickb** [comment]
@rogerwang Is there any hope of progress on this for macOS? I have a product that would benefit hugely from having this fixed, it's the only thing keeping me from switching to nw.js

**Comment by rogerwang** [comment]
@derrickb yes, will look into it. I believe the issue is mostly for Windows.

**Comment by derrickb** [comment]
@rogerwang thanks Roger! I've done extensive testing and unfortunately experienced the issue on Mac as well... tested with several nw.js versions last year, and again earlier this year.

**Comment by derrickb** [comment]
Appears related: #3299

**Comment by porsager** [comment]
This also happens on macos at least, so https://github.com/djipco/nwjs-j5-fix is still required :-/

Is it a won't-fix @rogerwang ?

**Comment by derrickb** [comment]
> This also happens on macos at least, so https://github.com/djipco/nwjs-j5-fix is still required :-/
> 
> Is it a won't-fix @rogerwang ?

When I tried nwjs-j5-fix on macOS, I still couldn't get stdin working. Were you able to?

**Comment by djipco** [comment]
@derrickb I'm using it on macOS Mojave with no issues.

**Comment by jacobsummers** [comment]
I am experiencing a similar (if not the same) issue on Mac in the latest builds of nwjs. process.stdout appears to work as expected but process.stdin does not. what i need to work is something like this  "echo foo | nw.js <myapp>", then in myappmain.js, process.stdin.read(3) should return "foo", instead it returns null. I have code that is working under node.js, leveraging "fs" and process.stdin.fd, but everything i have tried fails under nwjs.
